code/modules/mining/machinery/material_extractor.dm

```diff
@@ -0,0 +1,408 @@
+
+////////////////////////////////////////////////////
+// Actual machine
+////////////////////////////////////////////////////
+/obj/machinery/atmospherics/unary/material/extractor
+	name = "Gas extractor"
+	desc = "A machine for extracting liquids and gases from ices and hydrates."
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "extractor"
+	layer = MOB_LAYER+1 // Overhead
+	density = 1
+
+	atom_flags = ATOM_FLAG_NO_TEMP_CHANGE | ATOM_FLAG_NO_REACT //We wanna disable those to trigger reactions on our own terms
+	obj_flags = OBJ_FLAG_ANCHORABLE | OBJ_FLAG_ROTATABLE
+	use_power = POWER_USE_OFF
+	idle_power_usage = 25 //WATTS
+	power_rating = 1 KILOWATTS
+	power_channel = EQUIP
+	connect_types = CONNECT_TYPE_REGULAR
+
+	uncreated_component_parts = null
+	construct_state = /decl/machine_construction/default/panel_closed
+	required_interaction_dexterity = DEXTERITY_SIMPLE_MACHINES
+	public_variables = list(
+		/decl/public_access/public_variable/use_power,
+		/decl/public_access/public_variable/gas,
+		/decl/public_access/public_variable/pressure,
+		/decl/public_access/public_variable/temperature,
+		/decl/public_access/public_variable/reagents,
+		/decl/public_access/public_variable/reagents/volumes,
+		/decl/public_access/public_variable/reagents/free_space,
+		/decl/public_access/public_variable/reagents/total_volume,
+		/decl/public_access/public_variable/reagents/maximum_volume,
+		/decl/public_access/public_variable/material_extractor/has_bucket,
+	)
+	public_methods = list(
+		/decl/public_access/public_method/toggle_power,
+		/decl/public_access/public_method/material_extractor/flush_gas,
+		/decl/public_access/public_method/material_extractor/flush_reagents,
+	)
+	var/obj/item/chems/glass/output_container //#TODO: change this when plumbing is a thing
+	var/obj/input_holder/input_buffer //Since reagent_holder needs a parent object to exist on creation we gotta do this horrible hack
+
+/obj/machinery/atmospherics/unary/material/extractor/proc/get_output_loc()
+	return get_step(loc, get_output_dir())
+/obj/machinery/atmospherics/unary/material/extractor/proc/get_output_dir()
+	return turn(dir, 90)
+/obj/machinery/atmospherics/unary/material/extractor/proc/get_input_dir()
+	return turn(dir, -90)
+
+/obj/machinery/atmospherics/unary/material/extractor/Initialize(mapload, d = 0, populate_parts = TRUE)
+	. = ..()
+	if(populate_parts)
+		output_container = new/obj/item/chems/glass/bucket(src)
+	air_contents.volume = GAS_EXTRACTOR_GAS_TANK
+	if(!reagents)
+		create_reagents(GAS_EXTRACTOR_REAGENTS_TANK)
+	if(!input_buffer)
+		input_buffer = new(src)
+		input_buffer.create_reagents(GAS_EXTRACTOR_REAGENTS_INPUT_TANK) //Did this here because reimplementing that in the new() proc failed a test for some reasons
+	verbs |= /obj/machinery/atmospherics/unary/material/extractor/verb/FlushReagents
+	verbs |= /obj/machinery/atmospherics/unary/material/extractor/verb/FlushGas
```

Comment:
These will already be in the verb list for the machine.

Reply:
I thought so, but when I tested they weren't.

Reply:
That's weird, are you completely certain? I wonder if atmos machines are doing some weird verb stripping at some point for pipes or something.

Reply:
I'll run another test to see if it was just a one time thing...

Reply:
Yeah it works now. I had already removed the verbs from the init proc, so should be fine.